### ISSUE TITLE
Feature Request: separate Days in advance Reminders Settings

### ISSUE THREAD

**Issue by btittelbach** [mpcjanssen/simpletask-android#874]
Situation:
Currently there is only one slider to set the "Days in Advance" for reminders for both due and threshold dates.

Feature Request:
Please provide separate sliders for threshold and due date reminders so we can be reminded of threshold dates a different number of days in advance as of due dates. Also allow the threshold slider to be negative.

Why:
Threshold marks a date before which I don't want to bother with the todo. So Reminding me x days in advance of the threshold date is more annoying than useful. Setting a negative number od days to be reminded After the threshold date would be though!

Due Dates mark when the todo has to be finished. Often one does not want to be reminded of that on the last possible day but get advance warning. So I usually Set this to from 2 to 5 days. A number unsuitable for the threshold date.

**Comment by vojtechkral** [comment]
> Setting a negative number od days to be reminded After the threshold date would be though!

So, being reminded _after_ the threshold would be useful to you?

I agree having the same setting for both dues and thresholds doesn't make terribly much sense... I'll see what I can do...

**Comment by mpcjanssen** [comment]
@vojtechkral if it gets too complicated, I am planning on adding some sort of Lua callback for creating reminders. This will allow much more complex logic. So I would restrict this two making two sliders, one for each date type with a fallback to the current date setting.

**Comment by btittelbach** [comment]
@vojtechkral Yes being remined after the theshold date could be useful, but it's not important to me to warrant extra effort.
If we can just get two separate sliders where the minimum setting is 0, that would already be great! ;-)

**Comment by vojtechkral** [comment]
I'm not sure I'll be able to this right now, I'm a bit snowed under...

**Comment by mpcjanssen** [comment]
@vojtechkral I'll have a look at this.

**Comment by vojtechkral** [comment]
@mpcjanssen I can still do it, it just would probably take at least 1 week, as things have piled up for me (which the usage of Simpletask & GTD should prevent in the first place, but you know how it goes :-) )
